test/plugin/base_test.rb

```diff
@@ -338,6 +338,98 @@ def test_structured_payload_log
     end
   end
 
+  def test_autoformat_enabled_with_stackdriver_trace_id_as_trace
+    configs = [
+      APPLICATION_DEFAULT_CONFIG,
+      AUTOFORMAT_STACKDRIVER_TRACE_CONFIG
+    ]
+    configs.each do |test_config|
+      new_stub_context do
+        setup_gce_metadata_stubs
+        setup_logging_stubs do
+          d = create_driver(test_config)
+          d.emit(DEFAULT_TRACE_KEY => STACKDRIVER_TRACE_ID)
+          d.run
+          verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
+            expected_trace = \
+              "projects/#{d.instance.project_id}/traces/#{STACKDRIVER_TRACE_ID}"
+            assert_equal expected_trace, entry['trace'], 'stackdriver trace ' \
+                         'id should be autoformatted with config ' \
+                         "#{test_config}."
+          end
+        end
+      end
+    end
+  end
+
+  def test_autoformat_disabled_with_stackdriver_trace_id_as_trace
+    setup_gce_metadata_stubs
+    setup_logging_stubs do
+      d = create_driver(NO_AUTOFORMAT_STACKDRIVER_TRACE_CONFIG)
+      d.emit(DEFAULT_TRACE_KEY => STACKDRIVER_TRACE_ID)
+      d.run
+      verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
+        assert_equal STACKDRIVER_TRACE_ID, entry['trace'],
+                     'trace as stackdriver trace id should not be ' \
+                     'autoformatted with config ' \
+                     "#{NO_AUTOFORMAT_STACKDRIVER_TRACE_CONFIG}."
+      end
+    end
+  end
+
+  def test_no_trace_when_trace_key_not_exists_with_any_autoformat_config
+    configs = [
+      APPLICATION_DEFAULT_CONFIG,
+      AUTOFORMAT_STACKDRIVER_TRACE_CONFIG,
+      NO_AUTOFORMAT_STACKDRIVER_TRACE_CONFIG
+    ]
+    configs.each do |config|
+      new_stub_context do
+        setup_gce_metadata_stubs
+        setup_logging_stubs do
+          d = create_driver(config)
+          d.emit('msg' => log_entry(0))
+          d.run
+          verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
+            assert_nil entry['trace']
+          end
+        end
+      end
+    end
+  end
+
+  def test_non_stackdriver_trace_id_compliant_trace_with_any_autoformat_config
+    configs = [
+      APPLICATION_DEFAULT_CONFIG,
+      AUTOFORMAT_STACKDRIVER_TRACE_CONFIG,
+      NO_AUTOFORMAT_STACKDRIVER_TRACE_CONFIG
+    ]
+    traces = [
+      '',
```

Comment:
Hmm... The empty string case seems to have failed the unit test for gRPC. [Travis build](https://travis-ci.org/GoogleCloudPlatform/fluent-plugin-google-cloud/builds/433786673?utm_source=github_status&utm_medium=notification).

It's probably due to https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/test/plugin/test_out_google_cloud_grpc.rb#L439.

Reply:
use the function `assert_equal_with_default` to be able to verify even it's an empty string `''`